internal/reader/home_chain.go

```diff
@@ -112,19 +115,43 @@ func (r *homeChainPoller) poll() {
 }
 
 func (r *homeChainPoller) fetchAndSetConfigs(ctx context.Context) error {
-	var chainConfigInfos []ChainConfigInfo
-	err := r.homeChainReader.GetLatestValue(
-		ctx, "CCIPConfig", "getAllChainConfigs", nil, &chainConfigInfos,
-	)
-	if err != nil {
-		return err
+	var allChainConfigInfos []ChainConfigInfo
+	pageIndex := uint64(0)
+
+	for {
+		var chainConfigInfos []ChainConfigInfo
+		err := r.homeChainReader.GetLatestValue(
+			ctx,
+			"CCIPConfig",
+			"getAllChainConfigs",
+			map[string]interface{}{
+				"pageIndex": pageIndex,
+				"pageSize":  PageSize,
+			},
+			&chainConfigInfos,
+		)
+		if err != nil {
+			return err
+		}
+
+		if len(chainConfigInfos) == 0 {
+			break
+		}
+
+		allChainConfigInfos = append(allChainConfigInfos, chainConfigInfos...)
+		pageIndex++
+	}
+
+	if len(allChainConfigInfos) >= 0 {
```

Comment:
suggestion: the check is redundant - the length of an array will always be >= 0 - we can remove it

Reply:
removed redundant check